### PR TITLE
Added missing swagger for asset deletion

### PR DESCRIPTION
**Pull request by tiago-loureiro**

### CODE REVIEW

File: services/cargohold/src/CargoHold/API.hs [hunk 1]
```diff
@@ -167,6 +167,12 @@ sitemap = do
         header "Z-User"
         .&. capture "key"
 
+    document "DELETE" "deleteAsset" $ do
+        Doc.summary "Delete an asset"
+        Doc.parameter Doc.Path "key" Doc.bytes' $
+            Doc.description "Asset key"
+        Doc.response 200 "Asset deleted" Doc.end
```
Comment (neongreen) [inline services/cargohold/src/CargoHold/API.hs]: Please also add `Doc.errorResponse Error.assetNotFound` and `Doc.errorResponse Error.unauthorised`.

Comment (tiago-loureiro) [reply services/cargohold/src/CargoHold/API.hs]: > Please also add Doc.errorResponse Error.assetNotFound and Doc.errorResponse Error.unauthorised.

Well spotted, 👍

Comment (tiago-loureiro) [reply services/cargohold/src/CargoHold/API.hs]: Also added a few more error responses that I noticed were not documented https://github.com/wireapp/wire-server/commit/f2eb47f0cb4c549991bfb5ea65e2c69b2a4ce440

**Comment by neongreen** [comment]
The test failure seems spurious so I guess it can be merged:

```
    create 1-1 conversation between binding team members:        FAIL (0.26s)
      unexpected notification received: Notification {ntfId = 9490a89d-7bc2-
        11e8-8002-0a580a0209ae, ntfTransient = False, ntfPayload = List1 
        {toNonEmpty = fromList [("user",Object (fromList [("email",String 
        "success+0f38bd3b-9c81-4f2e-9f56-fc844687344a@simulator.amazonses.com"),
        ("locale",String "en"),("accent_id",Number 0.0),("picture",Array []),
        ("name",String "success+0f38bd3b-9c81-4f2e-9f56-fc844687344a
        @simulator.amazonses.com"),("id",String "df4248ea-95ef-4e7c-a7eb-
        faa826e81317"),("assets",Array [])])),("type",String "user.activate")] :| []}}
```

**Pull request merged**